src/server/getCurrentUser.ts

```diff
@@ -4,14 +4,35 @@ import { getSupabaseServerComponentClient } from "@/supabaseServer";
 import { logErrorReturnLogId } from "@/logger/logger";
 import { User } from "@supabase/gotrue-js";
 
-export async function getCurrentUser(): Promise<User | null> {
+type CurrentUserResponse =
+    | {
+          data: User;
+          error: null;
+      }
+    | {
+          data: null;
+          error: CurrentUserError;
+      };
+type CurrentUserErrorType = "userFetchFailed" | "noMatchingUsers";
+interface CurrentUserError {
+    type: CurrentUserErrorType;
+    logId: string;
+}
+
+export async function getCurrentUser(): Promise<CurrentUserResponse> {
     const supabase = getSupabaseServerComponentClient();
 
     const { data, error } = await supabase.auth.getUser();
 
     if (error) {
-        void logErrorReturnLogId("error with auth getUser", error);
+        const logId = await logErrorReturnLogId("error with auth getUser", error);
+        return { data: null, error: { type: "userFetchFailed", logId: logId } };
+    }
+
+    if (data === null) {
+        const logId = await logErrorReturnLogId("error with auth getUser: no matching users");
+        return { data: null, error: { type: "noMatchingUsers", logId: logId } };
     }
```

Comment:
Is this really needed? I think data is guaranteed to be not null if error is null?

Reply:
You're right, have removed